# HW 4
## Problem 2.1
Estimate time for benzene to completely evaporate from beaker assuming $N_A$ doesn't change with time.

Starting with a mole balance,
$$
\frac{d}{dt}\int c_idV = -\int N_i\cdot adS + \int S_idV \tag{1}
$$
which simplifies to
$$
\frac{d}{dt}\int c_AdV = -\int N_A\cdot adS \tag{1.1}
$$

$dV = A_cdh$ and $dh = -dz$ so ([1.1](#mjx-eqn-1.1)) goes to

$$
-A_c\frac{d}{dt}\int c_Adz = -\int N_A\cdot adS\tag{1.2}
$$

Assuming a negligible amount of air dissolved in the benzene, $c_A = \frac{\rho}{M}$ and is constant with respect to z so the left side of ([1.2](#mjx-eqn-1.2)) becomes
$$
-A_c\frac{\rho}{M}\frac{d}{dt}\int dz
$$

For the right hand side of ([1.2](#mjx-eqn-1.2)), $N_A$ is constant with respect to $S$ so the RHS becomes
$$
-N_AA_c \tag{1.3}
$$
now equating ([1.2](#mjx-eqn-1.2)) and ([1.3](#mjx-eqn-1.3)),
$$
\frac{\rho}{M}\frac{d}{dt}\int dz = N_A
$$

$N_A$ can be expanded to 
$$
N_A = x_AN +J_A \tag{2}
$$
assuming $N_B=0$ and applying Fick's Law ([2](#mjx-eqn-2)) is simplified to 
$$
N_A = x_AN_A -cD_{AB}\frac{dx_A}{dz}
$$
and rearanged to get
$$
N_A = -\frac{cD_{AB}}{1-x_A}\frac{dx_A}{dz}
$$
which can then be seperated and integrated,
$$
\int_{z_1}^{z_2}dz = -\frac{cD_{AB}}{N_A}\int_{x_{A_1}}^{x_{A_2}}\frac{dx_A}{1-x_A}
$$
$$
N_A = \frac{cD_{AB}}{z_2-z_1}\ln(\frac{1-x_{A_2}}{1-x_{A_1}}) \tag{2.1}
$$

$\therefore$
$$
\frac{\rho}{M}\frac{d}{dt}\int dz = \frac{cD_{AB}}{\Delta z}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})
$$
separating gives
$$
\int_0^t dt = \frac{\rho}{McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})}\int_{z_1}^{z_2}zdz
$$
where $z =\Delta z$
After integrating,
$$
t = \frac{\rho}{McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})}(\frac{z_2^2-z_1^2}{2}) \tag{3}
$$

The script below uses the following values to solve for t </br></br>
$M = 78.11 \frac{g}{mol}$</br></br>
$D_{AB} = 0.0905 \frac{cm^2}{sec}$</br></br>
$\rho = 0.876 \frac{g}{cm^3}$</br></br>
$z_1 = L$</br></br>
$z_2 = H$ where $L$ and $H$ are taken from the book to be 0.5cm and 6cm respectively.</br></br>
$x_{A_1}$ was calculated using Raoult's law: $x_A = \frac{p_A}{P} = \frac{0.12518}{1}$</br></br>
$T$ was assumed to be $298K$ and $P$ to be $1atm$ </br></br>
$c$ was calculated using the ideal gas law, $c = \frac{P}{RT}$

In [6]:
import numpy as np
M = 78.11                               #g/mol
P = 1                                               #atm
pA = .12518
DAB = .0905                                 #cm2/sec
ro = .876                                   #g/cm3
xa1 = pA/P
xa2 = 0
R = 8.2057*1000000                                   #cm3atm/mol/k
T = 273.15+25                                       #k
c = P/R/T                                           #mol/cm3
L = .5                                              #cm
H = 6                                                       #cm
t = (ro/M/c/DAB/np.log((1-xa2)/(1-xa1))*(H**2-L**2)*.5)/60/60/24
print(round(t),'days')

469007 days


It would take about 469007 days to evaporate if $N_A$ is constant in time

## Problem 2.2
Account for changing $N_A$ with liquid level

([2.1](#mjx-eqn-2.1)) still holds but the left side of the mole balance becomes
$$
\frac{d}{dt}c_AhA_c = c_AA_c\frac{dh}{dt} \tag{4}
$$

([3](#mjx-eqn-3)) can be rearanged to solve for $z_2$ as a function of time,
$$
z_2 = \sqrt{\frac{2McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})t}{\rho}+z_1^2} \tag{5}
$$
Given that $h = H-z$,
$$
h = H-\sqrt{\frac{2McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})t}{\rho}+z_1^2}
$$

$\therefore$
$$
\frac{dh}{dt} = -\frac{\frac{2McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})}{\rho}}{2\sqrt{\frac{2McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})}{\rho}t+z_1^2}}
$$



Putting it all back together gives
$$
-\frac{\frac{2McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})}{\rho}}{2\sqrt{\frac{2McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})}{\rho}t+z_1^2}}\frac{\rho}{M} = \frac{cD_{AB}}{z_2-z_1}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})
$$
which can be solved for t,
$$
t = -\frac{(\frac{\frac{2McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})}{\rho}}{2\frac{cD_{AB}}{z_2-z_1}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})}\frac{\rho}{M})^2-z_1^2}{\frac{2McD_{AB}\ln(\frac{1-x_{A_2}}{1-x_{A_1}})}{\rho}}
$$

In [7]:
a = 2*M*c*DAB*np.log((1-xa2)/(1-xa1))
b = c*DAB*np.log((1-xa2)/(1-xa1))/(H-L)
t = ((a*ro/M/2/b)**2 - L**2)/a
print(t/60/60/24)

343898.65074998315


Accounting for changing flux with liquid level, it would take about 343899 days to fully evaporate.</br></br>
This is less than if $N_A$ is assumed to be constant in time.</br></br>
I think that it is more accurate to account for changing $N_A$ because it does change with time and typically, the fewer assumptions that are made the more accurate the model.